src/axom/core/FlatMapView.hpp

```diff
@@ -201,12 +217,12 @@ class FlatMapView<KeyType, ValueType, IsConst, Hash>::IteratorImpl
 
   AXOM_HOST_DEVICE friend bool operator==(const IteratorImpl& lhs, const IteratorImpl& rhs)
   {
-    return lhs.m_internalIdx == rhs.m_internalIdx;
+    return (lhs.m_map.viewToSameMap(rhs.m_map) && lhs.m_internalIdx == rhs.m_internalIdx);
   }
 
   AXOM_HOST_DEVICE friend bool operator!=(const IteratorImpl& lhs, const IteratorImpl& rhs)
   {
-    return lhs.m_internalIdx != rhs.m_internalIdx;
+    return (!lhs.m_map.viewToSameMap(rhs.m_map) || lhs.m_internalIdx != rhs.m_internalIdx);
```

Comment:
To ensure consistency, should this just be 
```suggestion
    return lhs != rhs;
```